src/register.ts

```diff
@@ -82,9 +89,35 @@ export async function register(joy: JoyApi, account: string, handle: string, nam
     }
   }
 
+  // Check inviting members has invites
+  const hasInvites = await joy.invitingMemberHasInvites()
+  // Check inviting member has balance to top up new member account
+  const hasBalance = await joy.invitingMemberHasTopUpBalance()
+  // Check membership working group has budget
+  const workingGroupHasBudget = await joy.workingGroupHasBudget()
+
+  const canInviteMember = hasInvites && hasBalance && workingGroupHasBudget
+
+  if(!canInviteMember) {
+    // send email alert faucet is exhausted
+    sendEmailAlert(`Faucet is exhausted
+Inviting member has enough invites: ${hasInvites}
+Inviting member has enough topup balance: ${hasBalance}
+Members Working group has sufficient budget: ${workingGroupHasBudget}
+    `)
+
+    return callback('FaucetExhausted', 400)
+  }
+
+  // apply limit for global api call
+  const wasBlocked = await rollingLimiter.limit('register')
+  if (wasBlocked) {
+      return callback("TooManyRequests", 429);
+  }
```

Comment:
One more thing that came to my mind now is that perhaps there should be an additional alert in case the member registration got globally blocked, as this essentially renders the faucet unusable.

Reply:
I will add a log to console.